### PR TITLE
Add migration-tools.md

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
Add migration-tools.md

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/7480/files#diff-fd9a5f0b4ecb4683596d58a0d36ece853910907e96b6db43b46b0689c6a664c5R2
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: data-migration/migration-tools.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+tiup install dumpling tidb-lightning
+```
+
+> **Note:**
+>
+> To install a component of a specific version, use the `tiup install <component>[:version]` command.
+
+### Step 3. Update TiUP and its components (optional)
+
+It is recommended to see the update log and compatibility notes of the new version.
```
Comment (sunzhaoyang) [inline data-migration/migration-tools.md:94]:
```suggestion
It is recommended to see the release log and compatibility notes of the new version.
```

File: data-migration/migration-tools.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+The following diagram shows the user scenario of each migration tool.
+
+![TiDB Migration Tools](/media/migration-tools.png)
+
+The following table introduces the user scenarios, the supported upstreams and downstreams of migration tools.
+
+| Tool name | User scenario | Upstream (or the imported source file) | Downstream (or the output file) | Main advantage | Limitation |
+|:---|:---|:---|:---|:---|:---|
+|  [TiDB Data Migration (DM)](https://docs.pingcap.com/tidb-data-migration/stable/overview)| Data migration from MySQL-compatible databases to TiDB |  MySQL, MariaDB, Aurora, MySQL| TiDB   | <ul><li>A convenient and unified data migration task management tool that supports full data migration and incremental replication</li><li>Support filtering tables and operations</li><li>Support shard merge and migration</li></ul>  |  It is recommended to use DM to migrate full data with a size of less than 1 TB. |
```
Comment (sunzhaoyang) [inline data-migration/migration-tools.md:20]:
```suggestion
|  [TiDB Data Migration (DM)](https://docs.pingcap.com/tidb-data-migration/stable/overview)| Data migration from MySQL-compatible databases to TiDB |  MySQL, MariaDB, Aurora, MySQL| TiDB   | <ul><li>A convenient and unified data migration task management tool that supports full data migration and incremental replication</li><li>Support filtering tables and operations</li><li>Support shard merge and migration</li></ul>  | Data import speed is roughly the same as Lighting's tidb-backend, and much lower than Lighting's local-backend. So it is recommended to use DM to migrate full data with a size of less than 1 TB. |
```

**Review by sunzhaoyang — APPROVED** [review]
rest LGTM

File: data-migration/migration-tools.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+This document introduces the user scenarios, advantages, and limitations of these tools. You can choose the right tool according to your needs.
+
+The following diagram shows the user scenario of each migration tool.
+
+![TiDB Migration Tools](/media/migration-tools.png)
+
+The following table introduces the user scenarios, the supported upstreams and downstreams of migration tools.
+
+| Tool name | User scenario | Upstream (or the imported source file) | Downstream (or the output file) | Main advantage | Limitation |
```
Comment (hfxsd) [inline data-migration/migration-tools.md:18]:
```suggestion
| Tool name | User scenario | Upstream (or the imported source file) | Downstream (or the output file) | Advantages | Limitation |
```

File: data-migration/migration-tools.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+|:---|:---|:---|:---|:---|:---|
+|  [TiDB Data Migration (DM)](https://docs.pingcap.com/tidb-data-migration/stable/overview)| Data migration from MySQL-compatible databases to TiDB |  MySQL, MariaDB, Aurora, MySQL| TiDB   | <ul><li>A convenient and unified data migration task management tool that supports full data migration and incremental replication</li><li>Support filtering tables and operations</li><li>Support shard merge and migration</li></ul>  | Data import speed is roughly the same as that of TiDB Lighting's TiDB-backend, and much lower than that of TiDB Lighting's Local-backend. So it is recommended to use DM to migrate full data with a size of less than 1 TB. |
```
Comment (hfxsd) [inline data-migration/migration-tools.md:20]:
```suggestion
|  [TiDB Data Migration (DM)](https://docs.pingcap.com/tidb-data-migration/stable/overview)| Data migration from MySQL-compatible databases to TiDB |  MySQL, MariaDB, Aurora, MySQL| TiDB   | <ul><li>A convenient and unified data migration task management tool that supports full data migration and incremental replication</li><li>Support filtering tables and operations</li><li>Support shard merge and migration</li></ul>  | Data import speed is roughly the same as that of TiDB Lighting's TiDB-backend, and much lower than that of TiDB Lighting's Local-backend. So it is recommended to use DM to migrate full data with a size of less than 1 TiB. |
```

**Review by hfxsd — APPROVED** [review]

**Pull request merged**